source/server/hot_restarting_base.cc

```diff
@@ -224,5 +225,21 @@ std::unique_ptr<HotRestartMessage> HotRestartingBase::receiveHotRestartMessage(B
   return ret;
 }
 
+Stats::Gauge& HotRestartingBase::hotRestartGeneration(Stats::Scope& scope) {
```

Comment:
Do you think this gauge has any real use? I'm actually wondering with the test binary if we should just delete it and put a similar test gauge there?

Reply:
At the time we added it, I remember discussing whether we should change infrastructure to use this and drop the manually maintained 'epoch'. I'm fine punting this one and relying on epic, but it's up to you.

But one thing I want to avoid is making that change in the same PR where we fix a functional bug. WDYT?

Reply:
Sure that's fine can discuss in a different PR.